nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -753,14 +755,61 @@ export async function findUsableCommitmentsMutex(
   );
 }
 
+/**
+ *
+ * @function saveCommitments save a list of commitments in the database
+ * @param {[]} listOfCommitments a list of commitments to be saved in the database
+ * @returns if all the commitments in the list already exists in the database
+ * throw an error, else return a success message.
+ */
+export async function saveCommitments(listOfCommitments) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+
+  /**
+   * 1. listOfCommitments => get only the ids
+   */
+  const commitmentsIds = listOfCommitments.map(commitment => commitment._id);
```

Comment:
```suggestion
  const commitmentIds = listOfCommitments.map(commitment => commitment._id);
```

Reply:
In this case I prefer leave in plural. Because a list of ids for several commitments not just one.